Sources/RedisKit/Types/RedisSet.swift

```diff
@@ -0,0 +1,234 @@
+///     print(count) // Int(1)
+///
+/// - Note: Use of `.wait()` in the example is for simplicity.. Never call `.wait()` on a `NIO.EventLoop`!
+///
+/// - Important: `RedisSet<T>` instances have _reference_ semantics,
+///     as it holds a reference to a `RediStack.RedisClient` existential which could be a class.
+///
+///     It is also important to note that this will retain that instance in reference counts.
+///
+/// See [https://redis.io/topics/data-types-intro#sets](https://redis.io/topics/data-types-intro#sets)
+public struct RedisSet<Element> where Element: RESPValueConvertible {
+    /// The key in Redis that this instance is a reference to.
+    public var identifier: String { return self.id }
+    
+    private let id: String
+    private let client: RedisClient
+    
+    /// Initializes a new reference to a specific Redis key that holds a Set value type.
+    /// - Parameters:
+    ///     - identifier: The key identifier to reference this set.
+    ///     - client: The `RediStack.RedisClient` to use for making calls to Redis.
+    public init(identifier: String, client: RedisClient) {
+        self.id = identifier
+        self.client = client
+    }
+    
+    /// Resolves the number of elements in the set.
+    ///
+    /// See `RediStack.RedisClient.scard(of:)`
+    public var count: EventLoopFuture<Int> { return self.client.scard(of: self.id) }
+    /// Resolves a Boolean value that indicates whether the set is empty.
+    public var isEmpty: EventLoopFuture<Bool> { return self.count.map { $0 == 0 } }
+    /// Resolves all of elements in the set.
+    ///
+    /// All member elements will be converted into the type `Element`, based on its conformance to `RediStack.RESPValueConvertible`.
+    /// All `nil` values will be filtered from the result.
+    ///
+    /// See `RediStack.RedisClient.smembers(of:)`
+    public var allElements: EventLoopFuture<[Element]> {
+        return self.client.smembers(of: self.id)
+            .map { $0.compactMap(Element.init) }
+    }
+    
+    /// Resolves a Boolean value that indicates whether the given element exists in the set.
+    ///
+    /// See `RediStack.RedisClient.sismember(_:of:)`
+    /// - Parameter member: An element to look for in the set.
+    /// - Returns: A `NIO.EventLoopFuture<Bool>` resolving `true` if `member` exists in the set; otherwise, `false`.
+    public func contains(_ member: Element) -> EventLoopFuture<Bool> {
+        return self.client.sismember(member, of: self.id)
+    }
+}
+
+// MARK: Inserting Elements
+
+extension RedisSet {
+    /// Inserts the given element(s) in the set if it is not already present.
+    ///
+    /// See `RediStack.RedisClient.sadd(_:to:)`
+    /// - Parameter newMember: An element to insert into the set.
+    /// - Returns: A `NIO.EventLoopFuture<Bool>` resolving `true` if `newMember` was inserted into the set; otherwise, `false`.
+    @discardableResult
```

Comment:
Even though the `Bool` contained in the future is discardable, I don't think we can say the future itself is discardable. Ignoring it could lead to unexpected results. If only there were a way to indicate a future's result were discardable.

Reply:
Same with the other methods marked `@discardableResult`

Reply:
This is a fair argument - I've updated the code.